chall01/krusthol.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python3
+import sys
+import string
+
+def bad_character(str):
+    good = set(string.ascii_lowercase + ' ')
+    for c in str:
+        if c not in good:
+            return True
+    return False
+
+if len(sys.argv) != 2:
+    sys.exit("usage: ./krusthol.py <a-zA-Z string>")
+arg = sys.argv[1].lower()
+if len(arg) == 0 or bad_character(arg):
+    sys.exit("usage: ./krusthol.py <a-zA-Z string>")
+for c in arg:
+    if c == 'a':
+        sys.stdout.write('.-')
+    elif c == 'b':
+        sys.stdout.write('-...')
+    elif c == 'c':
+        sys.stdout.write('-.-.')
+    elif c == 'd':
+        sys.stdout.write('-..')
+    elif c == 'e':
+        sys.stdout.write('.')
+    elif c == 'f':
+        sys.stdout.write('..-.')
+    elif c == 'g':
+        sys.stdout.write('--.')
+    elif c == 'h':
+        sys.stdout.write('....')
+    elif c == 'i':
+        sys.stdout.write('..')
+    elif c == 'j':
+        sys.stdout.write('.---')
+    elif c == 'k':
+        sys.stdout.write('-.-')
+    elif c == 'l':
+        sys.stdout.write('.-..')
+    elif c == 'm':
+        sys.stdout.write('--')
+    elif c == 'n':
+        sys.stdout.write('-.')
+    elif c == 'o':
+        sys.stdout.write('---')
+    elif c == 'p':
+        sys.stdout.write('.--.')
+    elif c == 'q':
+        sys.stdout.write('--.-')
+    elif c == 'r':
+        sys.stdout.write('.-.')
+    elif c == 's':
+        sys.stdout.write('...')
+    elif c == 't':
+        sys.stdout.write('-')
+    elif c == 'u':
+        sys.stdout.write('..-')
+    elif c == 'v':
+        sys.stdout.write('...-')
+    elif c == 'w':
+        sys.stdout.write('.--')
+    elif c == 'x':
+        sys.stdout.write('-..-')
+    elif c == 'y':
+        sys.stdout.write('-.--')
+    elif c == 'z':
+        sys.stdout.write('--..')
+    elif c == ' ':
+        sys.stdout.write(' ')
```

Comment:
Here goes my main advice and a short introduction to "pythonic" way to write code in Python. 
First of all, instead of using numerous if-else statements, it's better to use dictionaries. Here is an example:
`my_awesome_dict = {'a': '.-', 'b': '-...'}`
Dictionary allows creating key-value pairs which is invaluably useful in so many cases. 
And here how you can quickly create one you need in pythonic way:
```
morse_input = ".- -..." # if you have multiple line string, you can create one in python using """ - triple quotes
morse_array = morse_input.split(" ") # again, if every value on new string, you can use "\n" instead of " ". And actually it's not array - in python it called list but it's kind of similar to C array
morse_dictionary = dict(zip(string.ascii_lowercase, morse_array) # this will create a dictionary in which every letter related to corresponding morse code
# or
morse_dictionary = {string.ascii_lowercase[i]: morse_array[i] for i in range(len(string.ascii_lowercase))}
# then add a space
morse_dictionary[" "] = " "
```
now we have a full dictionary
after that, all the if else statements can be truncated to one line:
`sys.stdout.write(morse_dictionary[c])`
even more than that! In Python, you can do the whole loop in one line by creating a string to print:
`print("".join(morse_dictionary[c] for c in arg))` 

And that is how laconic-pythonic Python is!